server/shared/src/test/scala/org/http4s/server/middleware/TimeoutSuite.scala

```diff
@@ -76,6 +81,18 @@ class TimeoutSuite extends Http4sSuite {
     }
   }
 
+  test(
+    "return a 503 error if the result takes too long and execute underlying uncancelable effect"
+  ) {
+    testMiddleware(5.milliseconds) { app =>
+      for {
+        _ <- app(uncancelableReq).map(_.status).assertEquals(Status.ServiceUnavailable)
+        _ <- checkStatus(app(uncancelableReq), Status.ServiceUnavailable)
+          .intercept[TimeoutException]
+      } yield ()
+    }
+  }
```

Comment:
Sorry, I'm a little confused what this test is demonstrating?

Reply:
Ah, thank you for explaining, I think I finally understood. The test for (1) is the `.intercept[TimeoutException]` because `1.seconds < IO.sleep(1100.milliseconds)` IIUC.

Reply:
Yeah, it so, you're correct.